docs(planning-design): add planning internal interface (#390)
autowarefoundation/autoware-documentation: docs/design/autoware-architecture/planning/index.md

## Component interface

> The following describes the input/output concept between Planning Component and other components. See the [Planning Component Interface](../../autoware-interfaces/components/planning.md) page for the current implementation.

Reviewer: ```suggestion
This section describes the inputs and outputs of the Planning Component and of its internal modules. See the [Planning Component Interface](../../autoware-interfaces/components/planning.md) page for the current implementation.
```
